new file mode 100755
index 0000000..61e71a8
--- /dev/null
+++ b/themes/stable/templates/views/views-view-field--name.html.twig
@@ -0,0 +1,24 @@
+{#
+/**
+ * @file
+ * Theme override for a single field in a view.
+ *
+ * Available variables:
+ * - view: The view that the field belongs to.
+ * - field: The field handler that can process the input.
+ * - row: The raw result of the database query that generated this field.
+ *
+ * When fetching output from the row this construct should be used:
+ * data = row[field.field_alias]
+ *
+ * The above will guarantee that you'll always get the correct data, regardless
+ * of any changes in the aliasing that might happen if the view is modified.
+ *
+ * @see template_preprocess_views_view_field()
+ */
+#}
+{% if view.id == 'user_admin_people' and view.current_display == 'page_1' %}
+  {% set uid = row.uid %}
+  {% set name = row._entity.name[0].value %}
+    {{ link(name, front_end_url ~ '/' ~ 'profile' ~ '?uid=' ~ uid ) }}
+{% endif %}
